### PR TITLE
feat(toolbar): adds toolbar component

### PR DESCRIPTION
**Pull request by mcoker**
fixes https://github.com/patternfly/patternfly-next/issues/959

This isn't ready yet. It needs the split button dropdown, options menu, and pagination component, which aren't verified yet.

Also here are some notes I took when talking to Gina about this component.

* buttons are 16px apart
* sort icon optional
* 32px between search icon and action groups if sort isn't there
* 8px margin on left and right of sort icon
* 8px margin to left of kebabi icon
* remove horizontal rule between pagination and left side
* search + kebab - 16px between
* sort + kebab - 8px between
* actions buttons collapse into kebab

### CODE REVIEW

**Comment by mcoker** [comment]
@matthewcarleton @christiemolloy this should be good to verify. @gdoyle1 @mcarrano updated the examples per our discussion.

**Comment by gdoyle1** [comment]
@mcoker Only update I see that needs to be made is when the Status filters are checked, "Filter" should say "Status"... other than that it looks great!

**Comment by mcoker** [comment]
@gdoyle1 thanks! updated.

**Comment by gdoyle1** [comment]
@mcoker You have the okay from me!! Thank you for your hard work on this, it looks amazing.

File: src/patternfly/components/Toolbar/toolbar.scss [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+.pf-c-toolbar,
+.pf-c-toolbar__filter-list,
+.pf-c-toolbar__action-group {
+  display: flex;
+  align-items: center;
+}
+
+// Bulk Select
+.pf-c-toolbar__bulk-select {
+  margin-right: var(--pf-c-toobar__bulk-select--MarginRight);
```
Comment (srambach) [inline src/patternfly/components/Toolbar/toolbar.scss]: Typo (toobar)

File: src/patternfly/components/Toolbar/toolbar.scss [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+.pf-c-toolbar__bulk-select {
+  margin-right: var(--pf-c-toolbar__bulk-select--MarginRight);
+}
+
+// Filter input group
+.pf-c-toolbar__filter-toggle {
+  --pf-c-toolbar--child--MarginLeft: var(--pf-c-toolbar__filter-toggle--MarginLeft);
+
+  &.pf-m-expanded {
+    color: var(--pf-c-toolbar__filter-toggle--m-expanded--Color);
```
Comment (matthewcarleton) [inline src/patternfly/components/Toolbar/toolbar.scss]: I think this is supposed to match line 20?

File: src/patternfly/components/Toolbar/toolbar.scss [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+  @media screen and (max-width: $pf-global--breakpoint--lg) {
+    display: none;
+
+    &.pf-m-expanded {
+      display: flex;
+    }
+  }
+
+  @media screen and (min-width: $pf-global--breakpoint--lg) {
+    --pf-c-toolbar__filter--MarginTop: 0;
```
Comment (matthewcarleton) [inline src/patternfly/components/Toolbar/toolbar.scss]: should the MQ be at the top?

File: src/patternfly/components/Toolbar/toolbar.scss [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    --pf-c-toolbar__filter--MarginBottom: 0;
```
Comment (matthewcarleton) [inline src/patternfly/components/Toolbar/toolbar.scss]: I don't see this being used.

**Comment by srambach** [comment]
I'm ok with this once the a11y check passes.

File: src/patternfly/components/Toolbar/examples/toolbar-expanded-example.hbs [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    {{#> pagination-nav}}
+      {{#> button button--modifier="pf-m-plain pf-m-disabled" button--attribute='aria-label="Go to first page" aria-disabled="true"'}}
+        {{> button-icon button-icon--type="angle-double-left"}}
+      {{/button}}
+      {{#> button button--modifier="pf-m-plain pf-m-disabled" button--attribute='aria-label="Go to previous page" aria-disabled="true"'}}
+        {{> button-icon button-icon--type="angle-left"}}
+      {{/button}}
+
+      {{#> pagination-nav-page-select pagination-nav-page-select--attribute='aria-label="Current page 1 of 4"'}}
+        <input class="pf-c-form-control" aria-label="Current page" type="number" min="1" max="4" size="2" value="1">
```
[the remark this answers is not in the document]
Comment (mcoker) [reply src/patternfly/components/Toolbar/examples/toolbar-expanded-example.hbs]: It works, updated to use `{{> form-control}}`

File: src/patternfly/components/Toolbar/toolbar.scss [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+  // Bulk Select
+  --pf-c-toolbar__bulk-select--MarginRight: var(--pf-global--spacer--md);
+
+  // Filter
+  --pf-c-toolbar__filter--MarginTop: calc(var(--pf-c-toolbar--PaddingBottom) + var(--pf-global--spacer--md));
+  --pf-c-toolbar__filter--MarginLeft: 0;
+
+  // Filter toggle toggle
+  --pf-c-toolbar__filter-toggle--MarginLeft: var(--pf-global--spacer--md);
+  --pf-c-toolbar__filter-toggle--m-expanded--c-button--Color: var(--pf-global--Color--100);
```
[the remark this answers is not in the document]
Comment (mcoker) [reply src/patternfly/components/Toolbar/toolbar.scss]: needs to change per your comment below

File: src/patternfly/components/Toolbar/toolbar.scss (hunk 2 above)
Comment (mcoker) [reply src/patternfly/components/Toolbar/toolbar.scss]: yup!

File: src/patternfly/components/Toolbar/toolbar.scss (hunk 3 above)
Comment (mcoker) [reply src/patternfly/components/Toolbar/toolbar.scss]: yep!

File: src/patternfly/components/Toolbar/toolbar.scss (hunk 4 above)
Comment (mcoker) [reply src/patternfly/components/Toolbar/toolbar.scss]: yep, margin between this and filter (chip) list is on the filter list now.

**Pull request merged**